<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>reg_testClass</class>
 <widget class="QDialog" name="reg_testClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>430</width>
    <height>560</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>430</width>
    <height>480</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>MS Shell Dlg 2</family>
    <pointsize>8</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>reg_test</string>
  </property>
  <widget class="QLabel" name="board_info">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>50</y>
     <width>410</width>
     <height>40</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>400</width>
     <height>40</height>
    </size>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Sunken</enum>
   </property>
   <property name="text">
    <string>TextLabel</string>
   </property>
   <property name="scaledContents">
    <bool>false</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QComboBox" name="tetridList">
   <property name="geometry">
    <rect>
     <x>140</x>
     <y>150</y>
     <width>180</width>
     <height>25</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>150</y>
     <width>40</width>
     <height>25</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text">
    <string>Tetrada:</string>
   </property>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>510</y>
     <width>410</width>
     <height>41</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="topMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QPushButton" name="exitButton">
      <property name="minimumSize">
       <size>
        <width>77</width>
        <height>25</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>77</width>
        <height>25</height>
       </size>
      </property>
      <property name="text">
       <string>Exit</string>
      </property>
      <property name="default">
       <bool>true</bool>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="fifo_info">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>250</y>
     <width>410</width>
     <height>40</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>400</width>
     <height>40</height>
    </size>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Sunken</enum>
   </property>
   <property name="text">
    <string>FIFO of tetrad</string>
   </property>
   <property name="scaledContents">
    <bool>false</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QGroupBox" name="statusGroup">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>180</y>
     <width>410</width>
     <height>61</height>
    </rect>
   </property>
   <property name="title">
    <string>Status</string>
   </property>
   <widget class="QLabel" name="hex_status">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>20</y>
      <width>60</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QLabel" name="hex_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>25</width>
      <height>30</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>HEX:</string>
    </property>
   </widget>
   <widget class="QLabel" name="bin_status">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>20</y>
      <width>150</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QLabel" name="bin_label">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>19</y>
      <width>30</width>
      <height>31</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>BIN:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="getButton">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>20</y>
      <width>75</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Get</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="RegGroup">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>300</y>
     <width>410</width>
     <height>101</height>
    </rect>
   </property>
   <property name="title">
    <string>Indirect Register Operations</string>
   </property>
   <widget class="QLabel" name="addr_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>Address (hex):</string>
    </property>
   </widget>
   <widget class="QPushButton" name="readButton">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>20</y>
      <width>75</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Read</string>
    </property>
   </widget>
   <widget class="QPushButton" name="writeButton">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>20</y>
      <width>75</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Write</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineAddr">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>20</y>
      <width>76</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0000</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineVal">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>60</y>
      <width>75</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0000</string>
    </property>
   </widget>
   <widget class="QLabel" name="val_label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>60</y>
      <width>60</width>
      <height>25</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>Value (hex):</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkRdWr">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>60</y>
      <width>120</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Cycling Read/Write</string>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="pld_txt">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>100</y>
     <width>410</width>
     <height>40</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>400</width>
     <height>40</height>
    </size>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Sunken</enum>
   </property>
   <property name="text">
    <string>PLD: none info</string>
   </property>
   <property name="scaledContents">
    <bool>false</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QComboBox" name="cbBoardList">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>10</y>
     <width>340</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>10</y>
     <width>40</width>
     <height>25</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="text">
    <string>Device:</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="RegGroup_2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>410</y>
     <width>410</width>
     <height>90</height>
    </rect>
   </property>
   <property name="title">
    <string>SPD Operations</string>
   </property>
   <widget class="QLabel" name="dev_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>50</width>
      <height>25</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>Dev (hex):</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineDev">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>20</y>
      <width>40</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0000</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineValSpd">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>52</y>
      <width>40</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0000</string>
    </property>
   </widget>
   <widget class="QLabel" name="val_label_spd">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>52</y>
      <width>60</width>
      <height>25</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>Value (hex):</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineNum">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>20</y>
      <width>40</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0000</string>
    </property>
   </widget>
   <widget class="QLabel" name="num_label">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>20</y>
      <width>60</width>
      <height>25</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>Num (hex):</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineReg">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>20</y>
      <width>40</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0000</string>
    </property>
   </widget>
   <widget class="QLabel" name="reg_label">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>20</y>
      <width>60</width>
      <height>25</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>Reg (hex):</string>
    </property>
   </widget>
   <widget class="QPushButton" name="readButtonSPD">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>50</y>
      <width>75</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Read</string>
    </property>
   </widget>
   <widget class="QPushButton" name="writeButtonSPD">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>50</y>
      <width>75</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Write</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>cbBoardList</tabstop>
  <tabstop>tetridList</tabstop>
  <tabstop>getButton</tabstop>
  <tabstop>lineAddr</tabstop>
  <tabstop>lineVal</tabstop>
  <tabstop>readButton</tabstop>
  <tabstop>writeButton</tabstop>
  <tabstop>checkRdWr</tabstop>
  <tabstop>lineDev</tabstop>
  <tabstop>lineNum</tabstop>
  <tabstop>lineReg</tabstop>
  <tabstop>lineValSpd</tabstop>
  <tabstop>readButtonSPD</tabstop>
  <tabstop>writeButtonSPD</tabstop>
  <tabstop>exitButton</tabstop>
 </tabstops>
 <resources>
  <include location="reg_test.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>exitButton</sender>
   <signal>clicked()</signal>
   <receiver>reg_testClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>252</x>
     <y>518</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tetridList</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>reg_testClass</receiver>
   <slot>updateFifoInfo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>189</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>getButton</sender>
   <signal>clicked()</signal>
   <receiver>reg_testClass</receiver>
   <slot>updateFifoInfo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>357</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>readButton</sender>
   <signal>released()</signal>
   <receiver>reg_testClass</receiver>
   <slot>ClickedRead()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>315</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>396</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>writeButton</sender>
   <signal>released()</signal>
   <receiver>reg_testClass</receiver>
   <slot>ClickedWrite()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>321</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>388</x>
     <y>419</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbBoardList</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>reg_testClass</receiver>
   <slot>updateDeviceInfo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>readButtonSPD</sender>
   <signal>released()</signal>
   <receiver>reg_testClass</receiver>
   <slot>ClickedReadSpd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>477</y>
    </hint>
    <hint type="destinationlabel">
     <x>363</x>
     <y>533</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>writeButtonSPD</sender>
   <signal>released()</signal>
   <receiver>reg_testClass</receiver>
   <slot>ClickedWriteSpd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>337</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateFifoInfo()</slot>
  <slot>ClickedRead()</slot>
  <slot>ClickedWrite()</slot>
  <slot>updateDeviceInfo()</slot>
  <slot>ClickedReadSpd()</slot>
  <slot>ClickedWriteSpd()</slot>
 </slots>
</ui>
